src/main/java/com/teamrocket/tms/services/user/UserServiceImpl.java

```diff
@@ -122,4 +121,25 @@ public TeamDTO createTeam(Long userId, TeamDTO teamDTO) {
 
         return teamService.createTeam(teamDTO);
     }
+
+    @Override
+    public TeamDTO assignTeamLeader(Long userId, Long teamId, Long leaderId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new UserNotFoundException("User with the id " + userId + " not found."));
+        log.info("User {} : {} retrieved. From assignTeamLeader.", userId, user.getLastName());
+        userServiceValidation.validateUserRoleCanPerformAction(user, Role.PROJECTMANAGER);
+
+        User leader = userRepository.findById(leaderId)
+                .orElseThrow(() -> new UserNotFoundException("User with the id " + leaderId + " not found."));
+        log.info("User {} : {} retrieved. From assignTeamLeader.", userId, user.getLastName());
+
+        userServiceValidation.validateAreUsersEquals(user, leader);
+
+        leader.setRole(Role.TEAMLEADER);
```

Comment:
Maybe we can do something about the case that a junior/medior is assigned as a team leader?